docs/overrides/main.html

```diff
@@ -3,5 +3,87 @@
+    posthog.init(
+        'phc_WEVBsfcilPS9u6e0CY22d8DqtFt0zalAP9A6iyCBHNH',
+        {
+            api_host: 'https://eu.i.posthog.com',
+            persistence: 'memory',  // cookieless tracking: https://posthog.com/tutorials/cookieless-tracking
+        }
+    );
+
+    document.addEventListener('DOMContentLoaded', function() {
+        var searchForm = document.querySelector('.md-search__form');
+        var searchInput = document.querySelector('[data-md-component="search-query"]');
+        var searchResults = document.querySelector('.md-search__output');
+        var lastSearchTerm = '';
+        var searchTimeout;
+        var lastSearchTime = 0;
+        const DEBOUNCE_MS = 1000; // Wait 1 second between typed searches
+
+        if (searchForm && searchInput && searchResults) {
+            // Track when search results appear from typing (debounced)
+            searchInput.addEventListener('input', function() {
+                clearTimeout(searchTimeout);
+                const searchTerm = this.value.trim();
+
+                // Don't track if term hasn't changed or is too short
+                if (searchTerm === lastSearchTerm || searchTerm.length < 3) {
+                    return;
+                }
+
+                searchTimeout = setTimeout(function() {
+                    const now = Date.now();
+                    // Only track if:
+                    // 1. Results exist
+                    // 2. Enough time has passed since last search
+                    if (searchResults.querySelector('.md-search-result__item') &&
+                        now - lastSearchTime > DEBOUNCE_MS) {
+                        posthog.capture('documentation_search', {
+                            search_term: searchTerm,
+                            search_type: 'typed'
+                        });
+                        lastSearchTerm = searchTerm;
+                        lastSearchTime = now;
+                    }
+                }, DEBOUNCE_MS);
+            });
+
+            // Track submitted searches
+            searchForm.addEventListener('submit', function(e) {
+                e.preventDefault();
```

Comment:
Keeps the search experience in MkDocs' client-side system (doesn't try and submit the form to a server)

Reply:
please add a comment to the code explaining in more detail. i don't fully understand, but it does look like it improves something, because when i try submitting multiple searches in the prod docs they don't work after the first.

Reply:
Added. Sadly this won't fix things in prod, it'll just make sure the JS doesn't interfere with existing mkdocs search fucntionality